src/loaders/FileLoader.js

```diff
@@ -55,225 +55,191 @@ class FileLoader extends Loader {
 
 		}
 
-		// Check for data: URI
-		const dataUriRegex = /^data:(.*?)(;base64)?,(.*)$/;
-		const dataUriRegexResult = url.match( dataUriRegex );
-		let request;
+		// Initialise array for duplicate requests
+		loading[ url ] = [];
+
+		loading[ url ].push( {
+			onLoad: onLoad,
+			onProgress: onProgress,
+			onError: onError,
+		} );
+
+		// https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/responseType
+		// if (this.responseType !== undefined) {
+		//   request.responseType = this.responseType;
+		// }
+
+		// https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/overrideMimeType
+		// if (request.overrideMimeType) {
+		//   request.overrideMimeType(
+		//     this.mimeType !== undefined ? this.mimeType : "text/plain",
+		//   );
+		// }
```

Comment:
Do you want to keep these?

Reply:
I'm a little confused by these actually. At the very least the translation into a [fetch request](https://developer.mozilla.org/en-US/docs/Web/API/fetch#parameters) is not quite clear.

Reply:
It's not obvious to me what `overrideMimeType`, either. I don't see the `setMimeType` function used anywhere in the repo aside from [this spot in MMDLoader](https://github.com/mrdoob/three.js/blob/0a4f02b86a98403cf63ec227bbcf77a05805a164/examples/jsm/loaders/MMDLoader.js#L313) which looks like it's just being used to set the charset to decode to. I tried messing with the STLLoader example by calling `setMimeType( 'text/html' )` to see if it had any effect or caused problems but it didn't do anything. Perhaps setting the charset to decode to is the only real usecase?

I found [this blog post](https://schneide.blog/2018/08/08/decoding-non-utf8-server-responses-using-the-fetch-api/) discussing issues with setting the charset with fetch and it seems that using `.text()` will always interpret files as UTF8. It looks like using `TextDecoder` is the modern way to deal with this issue? Are there other use cases for explicitly setting the override mime type @mrdoob?

Reply:
I don't remember the use cases... Maybe the best thing is to remove these and see if someone complains.

Reply:
> Maybe the best thing is to remove these and see if someone complains.

I like this approach 😁 Given that MMDLoader uses it to set the charset, though, should a new API be added to set that like `setCharset` or `setCharacterEncoding`? It's possible this could be parsed from the mimetype returned from the server so we can default to what the server says the charset should be but again this is something maybe we should wait to see if we get complaints to fix.

Given that MMDLoader seems to be the only available use case of `setMimeType` maybe @takahirox has some opinions?

Reply:
I don't remember really well but I guess the reason why I didn't use `TextDecoder` may be it was not stable when I implemented `MMDLoader`? Maybe we can try removing `setMimeType` and try `TextDecoder` instead?